docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+    },
+    {
+      "protected":       {
+      "alg": "ES256",
+      "extra": "yes",
+      "rz": "656f65b99da5d649ea315a52343add3642f14c7ff8d4ebce8ee33a2f4a4b41e0",
+      "typ": "CIC",
+      "upk": {
+          "alg": "ES256",
+          "crv": "P-256",
+          "kty": "EC",
+          "x": "PnzpEjQZ7bsCl2ZExs7dbFQlVzggv-_t50QuzZZWcoc",
+          "y": "1Z-xC6JZL2eAO57ovFJCstnBcMsOiqsGF1NJLyqq1F4"
+      }
+      },
+      "signature": "dsWL..."
+    }
+  ]
+}
+```
+
+#### GQ Signatures
+
+A [GQ (Guillou and Quisquater) signature](https://link.springer.com/content/pdf/10.1007/0-387-34799-2_16.pdf) is a a [Proof of Knowledge (PoK)](https://en.wikipedia.org/wiki/Proof_of_knowledge) of an RSA signature that keeps the RSA signature secret. It provides the same guarantee that the ID Token was signed by OpenID Provider, but it keeps the original RSA signature secret, preventing the ID Token in the PK Token from being used an ID Token. This works because a standard OpenID Connect-compliant service only accepts RSA signature and so will rejected an ID Token that has a GQ signature instead.
+
+Why do this? A non-GQ PK Token contains the ID Token signed and issued by the OP ( OpenID Provider). If the PK Token anyone who sees it could extract the ID Token including the signature issued by OP from the PK Token and then attempt to replay this ID Token to authenticate as the subject (remember ID Tokens are bearer authentication secrets). A correctly configured service would reject such a replayed ID Token because the audience (`aud`) value in the ID Token would not match the audience that the service expects. This is because ID Tokens issued for OpenPubkey will have a different audience than an ID Token issued for use in another service. Unfortunately, a common misconfiguration is that services do not check the audience claim in the ID Token. To prevent such replay attacks in OpenPubkey we use GQ signatures when a PK Token will be publicly posted.
+
+Note that in cases where a PK Token is merely used to authenticate to a server and is not made publicly available, no GQ signature is needed. For instance when OpenPubkey is used in SSH, SSH3, TLS and web authentication we do not need to use GQ signatures. When OpenPubkey is used to for software artifact signing where the signatures and PK Tokens will be posted to a public ledger, then GQ signatures or Zero Knowledge Proofs should be used.
+
+#### GQ Signature Protected Header
+
+When we replace an RSA signature in a PK Token with a GQ signature, we also replace the protected header of the RSA signature. GQ signatures not only provide a Proof-of-Knowledge of the original RSA Signature, but they also enable anyone who knows the original RSA signature to sign a message using the RSA signature as the signing key. Using this property the GQ signature acts a signature, signing the payload and the new protected header using the original RSA signature as the signing secret.
```

Comment:
Think this should be "a RSA" but I just had a solid debate with myself about it haha! Preference is to pick one even if it's not grammatically correct and stick with it

Reply:
I went with "an RSA"